saleor/product/utils/availability.py

```diff
@@ -12,35 +11,40 @@
 from ...extensions.manager import get_extensions_manager
 from .. import ProductAvailabilityStatus, VariantAvailabilityStatus
 
-ProductAvailability = namedtuple(
-    "ProductAvailability",
-    (
-        "available",
-        "on_sale",
-        "price_range",
-        "price_range_undiscounted",
-        "discount",
-        "price_range_local_currency",
-        "discount_local_currency",
-    ),
-)
-
-VariantAvailability = namedtuple(
-    "ProductAvailability",
-    (
-        "available",
-        "on_sale",
-        "price",
-        "price_undiscounted",
-        "discount",
-        "price_local_currency",
-        "discount_local_currency",
-    ),
-)
+if TYPE_CHECKING:
+    from django.db.models.query import QuerySet
+
+    from ...extensions.manager import ExtensionsManager
+
+
+@dataclass
+class ProductAvailability:
```

Comment:
![](https://i.giphy.com/media/J1S363SbQFselYWz68/giphy.webp)

At last we're getting rid of that namedtuple